src/libs/actions/SignInRedirect.js

```diff
@@ -41,12 +41,19 @@ function redirectToSignIn(errorMessage) {
             if (preferredLocale) {
                 Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, preferredLocale);
             }
-            if (errorMessage) {
-                Onyx.set(ONYXKEYS.SESSION, {error: errorMessage});
-            }
             if (activeClients && activeClients.length > 0) {
                 Onyx.set(ONYXKEYS.ACTIVE_CLIENTS, activeClients);
             }
+
+            const session = {
+                // We must set the authToken to null so that signOut action is triggered across other clients
+                authToken: null,
+            };
+
+            if (errorMessage) {
+                session.error = errorMessage;
+            }
+            Onyx.merge(ONYXKEYS.SESSION, session);
```

Comment:
Why not keep `Onyx.set` here, now that we combined `error` and `authToken` keys into one object?

Reply:
After, `Onyx.clear`, Onyx is initialized with the initial data https://github.com/Expensify/react-native-onyx/blob/6eadd7f73fca87e6551e43607ec78f645e17ef50/lib/Onyx.js#L618.

`set` here, was replacing that. So I used `merge`

Reply:
Aah nice, thanks for pointing that out! 👍

Reply:
NAB, it might be worth adding a brief comment here explaining why we're not using a `set()`. That explanation makes sense to me, but someone might have the innocent idea to change this from a `merge()` to a `set()` and break the intended behavior.

Reply:
Good point.

Reply:
Comment added.